File: src/main/java/kh/farrukh/aoc2024/day1/Day1A.java
package kh.farrukh.aoc2024.day1;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Comparator;

public class Day1A {

  public static void main(String[] args) throws IOException {
    BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

    var list1 = new ArrayList<Integer>();
    var list2 = new ArrayList<Integer>();

    var l = reader.readLine();

    while (l != null && !l.isEmpty()) {
      var ns = l.split("   ");
      list1.add(Integer.parseInt(ns[0]));
      list2.add(Integer.parseInt(ns[1]));
      l = reader.readLine();
    }

    list1.sort(Comparator.naturalOrder());
    list2.sort(Comparator.naturalOrder());

    var totalDist = 0;

    for (int i = 0; i < list1.size(); i++) {
      totalDist += Math.abs(list1.get(i) - list2.get(i));
    }

    System.out.println(totalDist);
  }

}

File: src/main/java/kh/farrukh/aoc2024/day1/Day1A.kt
package kh.farrukh.aoc2024.day1

import java.util.Scanner
import kotlin.math.abs


fun main() {
    val sc = Scanner(System.`in`)

    val list1 = mutableListOf<Int>()
    val list2 = mutableListOf<Int>()

    var line = sc.nextLine()

    while (line != null && line.isNotBlank()) {
        val nums = line.split("   ")
        list1.add(nums[0].toInt())
        list2.add(nums[1].toInt())

        line = sc.nextLine()
    }

    list1.sort()
    list2.sort()

    var totalDiff = 0

    for (i in 0 until list1.size) {
        val num1 = list1.elementAt(i)
        val num2 = list2.elementAt(i)
        val diff = abs(num1 - num2)
        totalDiff += diff
    }

    println(totalDiff)
}
